python/cudf/cudf/core/timestamp.py

```diff
@@ -0,0 +1,395 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+import numpy as np
+import pandas as pd
+
+from cudf.core.scalar import Scalar
+
+
+# TODO: Use cupy when it supports timestamps and timedeltas. See https://github.com/cupy/cupy/issues/2622
+# TODO: Replace with cudf.Timedelta. See https://github.com/rapidsai/cudf/issues/5882
+class Timestamp(Scalar):
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        pd_ts_kwargs = {k: v for k, v in kwargs.items() if k != "dtype"}
+        ts = pd.Timestamp(*args, **pd_ts_kwargs)
```

Comment:
Is there a reason you are constructing a pandas Timestamp here? Is it being done only for validation purposes?